core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java

```diff
@@ -753,6 +756,52 @@ private void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec,
     }
   }
 
+  private void unParseRegexpLike(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    SqlWriter.Frame ifFrame = writer.startFunCall("IF");
+    call.operand(0).unparse(writer, leftPrec, rightPrec);
+    writer.literal("rlike");
+    writer.print("r");
+    unParseRegexString(writer, call, leftPrec, rightPrec);
+    writer.print(",");
+    writer.literal("1");
+    writer.print(",");
+    writer.literal("0");
+    writer.endFunCall(ifFrame);
+  }
+
+  private void unParseRegexString(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    if (call.getOperandList().size() == 3) {
+      SqlCharStringLiteral modifiedRegexString = getModifiedRegexString(call);
+      modifiedRegexString.unparse(writer, leftPrec, rightPrec);
+    } else {
+      call.operand(1).unparse(writer, leftPrec, rightPrec);
+    }
+  }
+
+  private SqlCharStringLiteral getModifiedRegexString(SqlCall call) {
+    String matchArgument = call.operand(2).toString().replaceAll("'", "");
+    switch (matchArgument) {
+    case "i":
+      String caseInsensitiveLiteral = "(?i)";
+      String modifiedRegexForI = caseInsensitiveLiteral.concat(call.operand(1).toString()
+          .replaceAll("^'|'$", ""));
```

Comment:
Can you move this logic replaceAll("^'|'$", "") to a method called removeLeadingAndTrailingQuotes

Reply:
Done